src/main/java/org/sagebionetworks/web/client/widget/table/TableListWidget.java

```diff
@@ -155,7 +155,12 @@ private void setResults(List<EntityHeader> results) {
 		view.hideLoading();
 		for (EntityHeader header : results) {
 			addSynIdToList(header.getId());
-			view.addTableListItem(header);
+			TableEntityListGroupItem item = ginInjector.getTableEntityListGroupItem();
+			item.configure(header, event -> {
+				this.onTableClicked(header);
+			});
+			item.setItemCountVisible(shouldItemCountBeVisible());
+			view.addTableListItem(item);
```

Comment:
Before we would pass the header to the view and the view would configure the TableEntityListGroupItem widget.

Now we configure the TableEntityListGroupItem widget here and just pass it to the view.